### PR TITLE
Feature/Approve ToS

### LINKED ISSUE #112: Require User to Approve Terms and Conditions
Should approve to:
1. handle their personal data
2. ...

### PR DESCRIPTION
**Pull request by mblomdahl**
Translations on the way!

Resolves #112 .

### CODE REVIEW

File: tests/end2end/test_public_logging_in.py [hunk 1]
```diff
@@ -25,6 +25,46 @@ def test_can_log_in_returns_200(user, testapp):
...
 
 
+def test_user_must_approve_tos_on_login_if_unset(user, testapp):
+    """After login, the user is requested to approve ToS if not done already."""
+    user.tos_approved_at = None
+    user.save()
+    # Goes to pages that requires authentication and gets redirected to login page.
+    res = testapp.get('/collections/?foo=1&baz=2').follow()
+    # Fills out login form.
+    username_with_different_casing = user.email.upper()
```
Comment (mxtthias) [inline tests/end2end/test_public_logging_in.py]: Is there a special reason for this that we should document? I would assume it's from a copied test and thus covered elsewhere? (No need to change if that's the case.)

File: tests/forms/test_user_register.py [hunk 2]
```diff
@@ -10,17 +10,15 @@
 from xl_auth.user.forms import RegisterForm
 
 
-# noinspection PyUnusedLocal
-def test_validate_success(db, superuser):
+def test_validate_success(superuser):
```
Comment (mxtthias) [inline tests/forms/test_user_register.py]: Nice cleanup!

File: xl_auth/templates/users/approve_tos.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+{% extends "layout.html" %}
+{% block content %}
+    <div class="container-narrow">
+        <h1>{{ _('Approve ToS') }}</h1>
+        <p class="lead">
+            {{ _('For using this service you are required approve our Terms of Service.') }}
+        </p>
+        <p>{{ _('You agree that:') }}</p>
+        <ol>
+            <li>{{ _('Your name, email address, permission and other account details is visible to other users') }}</li>
```
Comment (mxtthias) [inline xl_auth/templates/users/approve_tos.html]: s/permission/permissions/

File: tests/end2end/test_public_logging_in.py (hunk 1 above)
Comment (mblomdahl) [reply tests/end2end/test_public_logging_in.py]: Not really.. Just trying to regularly emphasize that emails are not validated case sensitive. :)

Comment (mblomdahl) [reply tests/end2end/test_public_logging_in.py]: @mxtthias Line 19 above contains the same variable assignment with an inline comment. Should I remove the duplicate here?

File: messages.pot [hunk 4]
```diff
@@ -795,6 +816,40 @@ msgstr ""
 msgid "Edit User Details"
 msgstr ""
 
+#: xl_auth/templates/users/approve_tos.html:7
+msgid "For using this service you are required approve our Terms of Service."
```
Comment (mxtthias) [inline messages.pot]: s/For using/To use/

File: tests/end2end/test_public_logging_in.py (hunk 1 above)
Comment (mxtthias) [reply tests/end2end/test_public_logging_in.py]: Nah, I'd say we keep it as is. :)

**Pull request merged**